common/src/main/java/tachyon/TachyonURI.java

```diff
@@ -228,6 +231,50 @@ public TachyonURI getParent() {
   }
 
   /**
+   * Get the first n component of the TachyonURI path.
+   * There is no trailing separator as the path will be normalized by normalizePath() 
+   * 
+   * <pre>
+   * /a/b/c, 1              -> /a
+   * /a/b/c, 2              -> /a/b
+   * /a/b/c, 3              -> /a/b/c
+   * </pre>
+   * 
+   * @return the first n path component, null if the path does not n components
+   */
+  public String getPathComponent(int n) {
+    int numComp = getNumPathComponents();
+    if (numComp < n) {
+      return null;
+    } else if (numComp == n) {
+      return mUri.getPath();
+    } else {
+      String path = mUri.getPath();
+      String[] comp = path.split(SEPARATOR);
+      String[] nComp = new String[n];
+      for (int i = 0; i < n; i++) {
```

Comment:
format issue: `i ++`.

 Also, `"/a/b/c".split("/")` will return `["", "a", "b", "c"]`, so it should be `for (int i = 1; i <= n; i ++)`?